### PR TITLE
Plans section: Modernised header and navbar

### PR DESCRIPTION
**Pull request by chriskmnds**
### Proposed Changes

Partially addresses https://github.com/Automattic/martech/issues/1463

This brings the "modernized" header & navbar by default to `/plans/my-plan` & `/plan` (not behind a feature flag). This is another temp solution with follow-ups in TODO list below.

_Notes:_ Plans header & navigation are used for [jetpack-plans](https://github.com/Automattic/wp-calypso/tree/trunk/client/my-sites/plans/jetpack-plans) code, but the changes should not affect these (something for a follow-up)

#### Media

with some of the states I notice for the plans pages (info & upsell notices):

_Old `/plans`_

Screenshot 2023-02-06 at 4 25 02 PM

_New `/plans`_

Screenshot 2023-02-06 at 4 23 42 PM

_New `/plans` free site_

Screenshot 2023-02-06 at 5 12 03 PM

_`/plans/my-plan`_

Screenshot 2023-02-06 at 4 24 30 PM

_`/plans` domain-only flow_

Screenshot 2023-02-09 at 12 34 08 PM

### Testing Instructions

- Navigate to `/plans/[site_slug]?flags=onboarding/2023-pricing-grid`
- Confirm the changes
- Do the same without the flag `/plans/[site_slug]?flags=-onboarding/2023-pricing-grid`
- Do the same for domain-only flow (e.g. `plans/yearly/[freeplan.no.domain]?get_domain=freeplan.blog`)
- Confirm no changes in `jetpack-plans` (instructions [here](https://github.com/Automattic/wp-calypso/pull/72960#issuecomment-1420258817))

### TODO

- [x] Extract into a layout component to allow reuse | [issue](https://github.com/Automattic/wp-calypso/issues/73102)
- [x] Extract navigation into a component to allow reuse | [issue](https://github.com/Automattic/wp-calypso/issues/73102)
- [x] Consider navbar inside header to reduce some of the complexity | [issue](https://github.com/Automattic/wp-calypso/issues/73102)

### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to Automattic/martech#1463

### CODE REVIEW

**Comment by chriskmnds** [comment]
@Automattic/explorers added you as reviewers to please confirm the changes for the `domainSidebarExperimentUser` are sane. They should also be based on the modernized layout as that's the default for `/plans` pages after this PR. I did notice a slight weirdness when the sidebar collapsed, but that also happens in trunk (if it's urgent to fix, then please do it in a way to avoid creating further conflicts if possible: maybe here in this PR or after we merge this PR, assuming no bigger blockers 🙏 ):

Screenshot 2023-02-09 at 2 51 36 PM


(assuming I'm even understanding this flow correctly - I only forced `domainSidebarExperimentUser` to be truthy to test things).

**Comment by DustyReagan** [comment]
> @Automattic/explorers added you as reviewers to please confirm the changes for the domainSidebarExperimentUser are sane.

Thanks for the heads up! We walked through domain upsell flow with this PR and it seems sane. p1675958565293879-slack-CKZHG0QCR In theory the sidebar should not be visible (collapsed or full) during the domain upsell flow. We were unable to reproduce that, so I'm thinking maybe it was due to how you forced the upsell view? -- Just FYI, our plan is to launch that upsell experiment Friday so it's currently not yet public facing.

The domain upsell flow was the only bit I tested, but it gets a ✅ from me!

**Review by ddc22 — APPROVED** [review]
The domain flow for `/plans` looks a bit broken but that can be picked up later
image

If possible see if the mobile version can be fixed in a subsequent PR.

image


CC - @lucasmendes-design 

LGTM 🚢

**Comment by candy02058912** [comment]
> I did notice a slight weirdness when the sidebar collapsed

As what @DustyReagan mentioned above, I also tried testing around the PR.
I couldn't reproduce the case where the sidebar would show when in the domain upsell (experiment) flow.
It would be helpful if you could provide a bit more insight into how it was being tested. 🙏

**Comment by chriskmnds** [comment]
> The domain flow for /plans looks a bit broken but that can be picked up later

@jdc91 @lucasmendes-design It's not broken per se, just restyled. I purposefully adapted the new header to it as any clickthrough (in the navbar or sidebar) will throw the user into the new header style.

This is going to be refactored next anw, but I've made a quick change to make it look a little nicer (closer to the [new designs](https://github.com/Automattic/martech/issues/1491)). Code-wise pretty bad though.

Screenshot 2023-02-10 at 11 28 19 AM

Do we prefer this, or stick to what's in the branch right now? I wouldn't apply any further changes to this PR. If we are not happy with these two options, then this will need to be stepped back into a feature branch until the whole thing is complete. So not shipping anything right now.

**Comment by ddc22** [comment]
@chriskmnds 
> Do we prefer this

No please ship the approved changes for now and let's iterate from there.

**Comment by chriskmnds** [comment]
Thanks for confirming @Automattic/explorers :-)

> > I did notice a slight weirdness when the sidebar collapsed
> 
> As what @DustyReagan mentioned above, I also tried testing around the PR. I couldn't reproduce the case where the sidebar would show when in the domain upsell (experiment) flow. It would be helpful if you could provide a bit more insight into how it was being tested. 🙏

Hey @candy02058912 I'm guessing it's probably my scrappy method to test your changes. I only set [this prop](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans/main.jsx#L388) to `true` to surface the UI changes on the page (no other change). I still see the same thing on my end with a collapsed sidebar, but I haven't dug further. the "back" button is also not visible with the expanded sidebar.

Screenshot 2023-02-10 at 1 20 42 PM

**Comment by lucasmendes-design** [comment]
Hi @chriskmnds 

It's better but we don't need to show the CTA "Manage add-ons" for this flow, otherwise, it will take users out of the flow to buy the domain + plan. Can we change that from "Manage add-ons" to "My plan"? So, this "My plan" will do the same action as Skip button. 

image

**Comment by chriskmnds** [comment]
> Hi @chriskmnds
> 
> It's better but we don't need to show the CTA "Manage add-ons" for this flow, otherwise, it will take users out of the flow to buy the domain + plan. Can we change that from "Manage add-ons" to "My plan"? So, this "My plan" will do the same action as Skip button.
> 

Thanks, @lucasmendes-design. Agreed.
We shipped this as-is for now, and we'll be handling additional work on the headers in #73230. I believe you've covered everything we need for these parts in Automattic/martech#1491. will ping you for additional context/clarification later

**Comment by lucasmendes-design** [comment]
All good :)

**Comment by valterlorran** [comment]
Hey @chriskmnds, I think this PR caused the My Plan and Plans page to break for RTL languages. https://github.com/Automattic/wp-calypso/issues/73392

https://github.com/Automattic/wp-calypso/blob/b36bae0edb6d744105f1f7c9b5da40e757c433c2/client/my-sites/plans/modernized-layout.tsx#L138-L140

**Comment by valterlorran** [comment]
I opened this PR(https://github.com/Automattic/wp-calypso/pull/73444) I'd love your 👀 on it!

**Pull request merged**
